components/raftstore/src/store/peer.rs

```diff
@@ -1681,6 +1775,32 @@ impl Peer {
         None
     }
 
+    pub fn propose_batch_request<T, C>(&mut self, ctx: &mut PollContext<T, C>) {
+        if let Some((req, cb)) = self.batch_req_builder.build(&mut ctx.raft_metrics.propose) {
+            if let Err(e) = util::check_term(&req, self.term()) {
+                let mut resp = RaftCmdResponse::default();
+                cmd_resp::bind_error(&mut resp, e);
+                cb.invoke_with_response(resp);
+                return;
+            }
+            match self.propose_normal(ctx, req) {
```

Comment:
It's not the same as original execution path. Hibernated region needs to be notified. I worry separate execution paths can lead to bug easily.